pkg/agents/team6/treaties.go

```diff
@@ -0,0 +1,304 @@
+		if conditionCheck || conditionValue < levels.strongLevel {
+			return consultUtility
+		}
+		return true
+	default:
+		return consultUtility
+	}
+}
+
+// Decides if to accept or reject a treaty using utility. Used in "considerUtility" function above
+func (a *CustomAgent6) considerTreatyUsingUtility(t *messages.Treaty) bool {
+
+	// Log
+
+	// 1. Estimate the food intake of the proposed treaty
+
+	// Calculates how much food has been available on average on this floor
+	sum := food.FoodType(0)
+	for _, food := range a.shortTermMemory {
+		sum += food
+	}
+	averageFoodAvailable := float64(sum) / math.Max(float64(len(a.shortTermMemory)), 1.0)
+
+	// Converts different treaty "Request types" to a "food intake"
+	estimatedTakeFood := a.convertToTakeFoodAmount(averageFoodAvailable, t.Request(), t.RequestValue())
+	if estimatedTakeFood == -1 {
+		return false
+	}
+
+	// Checks the exact request condition. Only consider meaningful treaties (GE, GT)
+	if t.RequestOp() == messages.GE || t.RequestOp() == messages.GT /*t.ConditionOp() == messages.EQ || */ {
+		// The treaty is of the form "Take X (or less) food"
+
+		// 2. Calculate the agent's utility given different outcomes (accept or reject treaty)
+
+		// a. estimated expected utility when accepting the treaty
+		treatyTrustFactor := 1.0
+		treatyUtility := treatyTrustFactor * calculateUtility(float64(estimatedTakeFood), float64(a.desiredFoodIntake()), a.currBehaviour.string())
+
+		// b. estimated utility when rejecting the treaty
+		currentShortTermUtility := a.evaluateUtility(a.shortTermMemory) // estimated utility on the current floor
+		currentLongTermUtility := a.evaluateUtility(a.longTermMemory)   // estimated utility over the entire time in the tower
+
+		// benefit of signing the treaty (> 0 == beneficial)
+		shortTermBenefit := treatyUtility - currentShortTermUtility
+		longTermBenefit := treatyUtility - currentLongTermUtility
+
+		estimatedPeriod := int(a.reassignPeriodGuess)
+		estimatedTimeLeft := estimatedPeriod - len(a.shortTermMemory)
+		benefit := 0.0
+
+		// only think in the short term if
+		// - the duration is shorter than the time left in estimated reassignment period OR
+		// - we only have short term experience OR
+		// - HP is critical (survival instincts take over)
+		if t.Duration() < estimatedTimeLeft ||
+			len(a.shortTermMemory) == len(a.longTermMemory) ||
+			a.HP() <= a.HealthInfo().HPCritical {
+
+			benefit = shortTermBenefit
+
+		} else {
+			// → The longer the duration of a treaty, the more important is it's long term benefit
+			// E.g.
+			// (1) time left on this level = 4, duration = 5 --> shortTermFocus 80%
+			// 		--> The treaty mostly matters in the short term
+			// (2) time left on this level = 4, duration = 8 --> shortTermFocus 50%
+			// 		--> The treaty counts as much on this level as for the time after
+			// (3) time left on this level = 1, duration = 100 --> shortTermFocus 1%
+			// 		--> The treaty mostly matters in the long term
```

Comment:
optional, but maybe consider whether these comments are necessary here

Reply:
We'll leave it